### PR TITLE
Add libxpm/3.5.13

### PR DESCRIPTION
**Pull request by friendlyanon**
Specify library name and version:  **libxpm/3.5.13**

Issue: #8618  
The library's [Microsoft README](https://gitlab.freedesktop.org/xorg/lib/libxpm/-/blob/master/doc/README.MSW) says to DIY the build files, so this time I was forced to make a CML anyway.

@cmb69 this recipe does not bundle the X11 headers like your distribution. Do you think that's a problem? Could it be solved in `ext/gd` instead?

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by cmb69** [comment]
> this recipe does not bundle the X11 headers like your distribution. Do you think that's a problem?

We could probably provided these headers by other means.

**Comment by friendlyanon** [comment]
I added an option to vendor the Xorg headers.

**Comment by friendlyanon** [comment]
Well, I guess I'll have to revert these new commits, Xorg really is needed, not just the headers.

**Comment by cmb69** [comment]
> Well, I guess I'll have to revert these new commits, Xorg really is needed, not just the headers.

I assume it's needed for other platforms than Windows.

**Comment by friendlyanon** [comment]
Yes, the log posted by the bot shows that there are undefined references. Only headers being used would have been too nice to be true.

**Comment by friendlyanon** [comment]
I don't know how to deal with #8790 so I just left MacOS disabled in this recipe.

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: recipes/libxpm/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    no_copy_source = True
+
+    def validate(self):
+        if self.settings.os not in ("Windows", "Linux", "FreeBSD"):
+            raise ConanInvalidConfiguration(
+                "libXpm is supported only on Windows, Linux and FreeBSD"
+            )
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True)
```
Comment (prince-chrismc) [inline recipes/libxpm/all/conanfile.py:37]: Typically we prefer out of source builds

https://github.com/conan-io/conan-center-index/blob/9f21b791bce36c330be3fc06dd4c6a8aaa5cf3a5/recipes/mfast/all/conanfile.py#L91-L93

**Pull request merged**